### PR TITLE
Feature: webpack cli task

### PR DESCRIPTION
**Pull request by anikethsaha**
**PR type ?**
Feature

**Did you add tests for your changes?**
No

**If relevant, did you update the documentation?**
added commented docs

**Summary**
https://github.com/microsoft/just/issues/168
As of now added only the `webpack-cli init` command 

**Does this PR introduce a breaking change?**
No

**Other information**
NA

### CODE REVIEW

**Comment by anikethsaha** [comment]
cc @kenotron

File: packages/just-scripts/src/tasks/webpackCLITask.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ * @param  {string} customScaffold? - to pass any webpack-scaffold
+ * @param  {Boolean=false} auto - to pass the --auto flag, which will generate a default webpack.config.js
+ * @returns TaskFunction
+ */
+export function webpackCliInitTask(
+  customScaffold?: string,
+  auto: Boolean = false
+): TaskFunction {
+  return async function webpackCli() {
+    const init = tryRequire("@webpack-cli/init").default;
```
Comment (kenotron) [inline packages/just-scripts/src/tasks/webpackCLITask.ts]: Is this a different one than https://www.npmjs.com/package/webpack-cli ?

**Comment by kenotron** [comment]
Thanks for the feature!

File: packages/just-scripts/src/tasks/webpackCLITask.ts (hunk 1 above)
Comment (anikethsaha) [reply packages/just-scripts/src/tasks/webpackCLITask.ts]: No it's the same.  This is the implementation of webpack cli init command

File: packages/just-scripts/src/tasks/webpackCLITask.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if (!init) {
+      logger.warn(
+        "webpack-cli init requires three dependencies: @webpack-cli/init (preferred - as a devDependency )"
```
Comment (kenotron) [inline packages/just-scripts/src/tasks/webpackCLITask.ts]:
```suggestion
        "webpack-cli init requires three dependencies: @webpack-cli/init (preferred - as a devDependency)"
```

File: packages/just-scripts/src/tasks/webpackCLITask.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      );
+      return;
+    }
+    logger.info(`Running Webpack-cli init `);
+    if (typeof customScaffold === "undefined") {
+      if (auto) {
+        return new Promise((resolve, reject) => {
+          try {
+            init(null, null, null, "--auto");
```
Comment (kenotron) [inline packages/just-scripts/src/tasks/webpackCLITask.ts]: This looks like a synchronous thing, you don't really need to return Promise in that case, right?

Comment (anikethsaha) [reply packages/just-scripts/src/tasks/webpackCLITask.ts]: the generator running seems async . I will take a bit look at this and will refactor the commit then .
May I think its sync , we done need promise then !

Comment (kenotron) [reply packages/just-scripts/src/tasks/webpackCLITask.ts]: ping

Comment (anikethsaha) [reply packages/just-scripts/src/tasks/webpackCLITask.ts]: Sorry for the delay,  I will try to resolve this by this weekend :+1:

Comment (anikethsaha) [reply packages/just-scripts/src/tasks/webpackCLITask.ts]: @kenotron Done 👍 
Changed it to sync !.

**Comment by anikethsaha** [comment]
@kenotron ping !

**Comment by kenotron** [comment]
Okay, finally got a chance to play around with this PR. This one task requires the consumer to have a few packages in their deps - in this case:

`webpack-cli` as well as `@webpack-cli/init` - if this were a simple spawn of the `webpack-cli` CLI (like the other tools in my tasks), it seems that it would simplify the flow a bit. Want to retool this task to forward a call to `webpack-cli` via CLI instead?

**Comment by kenotron** [comment]
> Okay, finally got a chance to play around with this PR. This one task requires the consumer to have a few packages in their deps - in this case:
> 
> `webpack-cli` as well as `@webpack-cli/init` - if this were a simple spawn of the `webpack-cli` CLI (like the other tools in my tasks), it seems that it would simplify the flow a bit. Want to retool this task to forward a call to `webpack-cli` via CLI instead?

I might be missing something too, maybe you can share what the advantage of calling through an API would be?

**Comment by anikethsaha** [comment]
>  maybe you can share what the advantage of calling through an API would be?

I don't think any major benefits of either of the choice. One thing which I think it may concern is that as `webpack-cli` is going to v4 it may change the integrating of the `init` package in it!  so maybe output response may change but it can be changed again it won't be a much of work may be just 2-3 lines of code changes. As `webpack-cli/init` is kind of standalone project and they are having API exposed to use it so I thought of using them in first place. 
I would love to get the opinion of @evenstensberg and @ematipico on which one can be better to implement. I guess user may need `webpack`, `webpack-cli`, and `webpack-cli/init` as a (dev)dependency in both the cases

**Comment by kenotron** [comment]
> I guess user may need `webpack`, `webpack-cli`, and `webpack-cli/init` as a (dev)dependency in both the cases

I see - if the deps requirement is the same, then I'll merge this!

**Comment by kenotron** [comment]
MERGED!

**Pull request merged**